File: app/src/main/java/com/arejas/dashboardofthings/data/helpers/DataHelper.java
package com.arejas.dashboardofthings.data.helpers;

import com.arejas.dashboardofthings.utils.Enumerators;

public class DataHelper {

    public static Enumerators.NotificationState getNotificationStatus(String dataReceived,
                                                                      Enumerators.DataType dataType,
                                                                      Float thresholdAboveWarningRaw,
                                                                      Float thresholdAboveCriticalRaw,
                                                                      Float thresholdBelowWarningRaw,
                                                                      Float thresholdBelowCriticalRaw,
                                                                      String thresholdEqualsWarningString,
                                                                      String thresholdEqualsCriticalString) {
        try {
            if (dataType != null) {
                if (dataType.equals(Enumerators.DataType.INTEGER)) {
                    int thresholdAboveWarning = (thresholdAboveWarningRaw != null) ?
                            thresholdAboveWarningRaw.intValue() : Integer.MAX_VALUE;
                    int thresholdAboveCritical = (thresholdAboveCriticalRaw != null) ?
                            thresholdAboveCriticalRaw.intValue() : Integer.MAX_VALUE;
                    int thresholdBelowWarning = (thresholdBelowWarningRaw != null) ?
                            thresholdBelowWarningRaw.intValue() : Integer.MIN_VALUE;
                    int thresholdBelowCritical = (thresholdBelowCriticalRaw != null) ?
                            thresholdBelowCriticalRaw.intValue() : Integer.MIN_VALUE;
                    int intValue = Integer.valueOf(dataReceived);
                    if ((intValue >= thresholdAboveCritical) && (intValue <= thresholdBelowCritical)) {
                        return Enumerators.NotificationState.CRITICAL;
                    } else if ((intValue >= thresholdAboveWarning) && (intValue <= thresholdBelowWarning)) {
                        return Enumerators.NotificationState.WARNING;
                    } else {
                        return Enumerators.NotificationState.NONE;
                    }
                } else if (dataType.equals(Enumerators.DataType.DECIMAL)) {
                    float thresholdAboveWarning = (thresholdAboveWarningRaw != null) ?
                            thresholdAboveWarningRaw : Float.MAX_VALUE;
                    float thresholdAboveCritical = (thresholdAboveCriticalRaw != null) ?
                            thresholdAboveCriticalRaw : Float.MAX_VALUE;
                    float thresholdBelowWarning = (thresholdBelowWarningRaw != null) ?
                            thresholdBelowWarningRaw : Float.MIN_VALUE;
                    float thresholdBelowCritical = (thresholdBelowCriticalRaw != null) ?
                            thresholdBelowCriticalRaw : Float.MIN_VALUE;
                    float floatValue = Float.valueOf(dataReceived);
                    if ((floatValue >= thresholdAboveCritical) && (floatValue <= thresholdBelowCritical)) {
                        return Enumerators.NotificationState.CRITICAL;
                    } else if ((floatValue >= thresholdAboveWarning) && (floatValue <= thresholdBelowWarning)) {
                        return Enumerators.NotificationState.WARNING;
                    } else {
                        return Enumerators.NotificationState.NONE;
                    }
                } else if (dataType.equals(Enumerators.DataType.BOOLEAN)) {
                    Boolean thresholdEqualsWarning = ((thresholdEqualsWarningString != null) &&
                            (!thresholdEqualsWarningString.isEmpty())) ?
                            Boolean.valueOf(thresholdEqualsWarningString) : null;
                    Boolean thresholdEqualsCritical = ((thresholdEqualsCriticalString != null) &&
                            (!thresholdEqualsCriticalString.isEmpty())) ?
                            Boolean.valueOf(thresholdEqualsCriticalString) : null;
                    boolean booleanValue = Boolean.valueOf(dataReceived);
                    if ((thresholdEqualsCritical != null) && (booleanValue == thresholdEqualsCritical.booleanValue())) {
                        return Enumerators.NotificationState.CRITICAL;
                    } else if ((thresholdEqualsWarning != null) && (booleanValue == thresholdEqualsWarning.booleanValue())) {
                        return Enumerators.NotificationState.WARNING;
                    } else {
                        return Enumerators.NotificationState.NONE;
                    }
                } else {
                    if ((thresholdEqualsCriticalString != null) && (dataReceived.equals(thresholdEqualsCriticalString))) {
                        return Enumerators.NotificationState.CRITICAL;
                    } else if ((thresholdEqualsWarningString != null) && (dataReceived.equals(thresholdEqualsWarningString))) {
                        return Enumerators.NotificationState.WARNING;
                    } else {
                        return Enumerators.NotificationState.NONE;
                    }
                }
            }
            return null;
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }

}

File: app/src/main/java/com/arejas/dashboardofthings/data/helpers/DataHelper.kt
package com.arejas.dashboardofthings.data.helpers

import com.arejas.dashboardofthings.utils.Enumerators

object DataHelper {

    fun getNotificationStatus(
        dataReceived: String,
        dataType: Enumerators.DataType?,
        thresholdAboveWarningRaw: Float?,
        thresholdAboveCriticalRaw: Float?,
        thresholdBelowWarningRaw: Float?,
        thresholdBelowCriticalRaw: Float?,
        thresholdEqualsWarningString: String?,
        thresholdEqualsCriticalString: String?
    ): Enumerators.NotificationState? {
        try {
            if (dataType != null) {
                if (dataType == Enumerators.DataType.INTEGER) {
                    val thresholdAboveWarning =
                        thresholdAboveWarningRaw?.toInt() ?: Integer.MAX_VALUE
                    val thresholdAboveCritical =
                        thresholdAboveCriticalRaw?.toInt() ?: Integer.MAX_VALUE
                    val thresholdBelowWarning =
                        thresholdBelowWarningRaw?.toInt() ?: Integer.MIN_VALUE
                    val thresholdBelowCritical =
                        thresholdBelowCriticalRaw?.toInt() ?: Integer.MIN_VALUE
                    val intValue = Integer.valueOf(dataReceived)
                    return if (intValue >= thresholdAboveCritical || intValue <= thresholdBelowCritical) {
                        Enumerators.NotificationState.CRITICAL
                    } else if (intValue >= thresholdAboveWarning || intValue <= thresholdBelowWarning) {
                        Enumerators.NotificationState.WARNING
                    } else {
                        Enumerators.NotificationState.NONE
                    }
                } else if (dataType == Enumerators.DataType.DECIMAL) {
                    val thresholdAboveWarning =
                        thresholdAboveWarningRaw ?: java.lang.Float.MAX_VALUE
                    val thresholdAboveCritical =
                        thresholdAboveCriticalRaw ?: java.lang.Float.MAX_VALUE
                    val thresholdBelowWarning =
                        thresholdBelowWarningRaw ?: java.lang.Float.MIN_VALUE
                    val thresholdBelowCritical =
                        thresholdBelowCriticalRaw ?: java.lang.Float.MIN_VALUE
                    val floatValue = java.lang.Float.valueOf(dataReceived)
                    return if (floatValue >= thresholdAboveCritical || floatValue <= thresholdBelowCritical) {
                        Enumerators.NotificationState.CRITICAL
                    } else if (floatValue >= thresholdAboveWarning || floatValue <= thresholdBelowWarning) {
                        Enumerators.NotificationState.WARNING
                    } else {
                        Enumerators.NotificationState.NONE
                    }
                } else if (dataType == Enumerators.DataType.BOOLEAN) {
                    val thresholdEqualsWarning =
                        if (thresholdEqualsWarningString != null && !thresholdEqualsWarningString.isEmpty())
                            java.lang.Boolean.valueOf(thresholdEqualsWarningString)
                        else
                            null
                    val thresholdEqualsCritical =
                        if (thresholdEqualsCriticalString != null && !thresholdEqualsCriticalString.isEmpty())
                            java.lang.Boolean.valueOf(thresholdEqualsCriticalString)
                        else
                            null
                    val booleanValue = java.lang.Boolean.valueOf(dataReceived)
                    return if (thresholdEqualsCritical != null && booleanValue == thresholdEqualsCritical) {
                        Enumerators.NotificationState.CRITICAL
                    } else if (thresholdEqualsWarning != null && booleanValue == thresholdEqualsWarning) {
                        Enumerators.NotificationState.WARNING
                    } else {
                        Enumerators.NotificationState.NONE
                    }
                } else {
                    return if (thresholdEqualsCriticalString != null && dataReceived == thresholdEqualsCriticalString) {
                        Enumerators.NotificationState.CRITICAL
                    } else if (thresholdEqualsWarningString != null && dataReceived == thresholdEqualsWarningString) {
                        Enumerators.NotificationState.WARNING
                    } else {
                        Enumerators.NotificationState.NONE
                    }
                }
            }
            return null
        } catch (e: Exception) {
            e.printStackTrace()
            return null
        }

    }

}
